src/main/java/org/isf/lab/gui/LabEdit.java

```diff
@@ -351,47 +349,33 @@ private JCheckBox getInPatientCheckBox() {
 	 * If no patient is chosen only Name, Age and Sex will be saved
 	 * in LABORATORY table (Name can be empty)
 	 */
-	private JComboBox getPatientComboBox() {
+	private PatientComboBox getPatientComboBox() {
 		if (patientComboBox == null) {
-			patientComboBox = new JComboBox();
-			patSelected=null;
+
 			PatientBrowserManager patBrowser = Context.getApplicationContext().getBean(PatientBrowserManager.class);
 			ArrayList<Patient> pat = null;
 			try {
 				pat = patBrowser.getPatient();
 			} catch (OHServiceException e) {
 				OHServiceExceptionUtil.showMessages(e);
 			}
-			patientComboBox.addItem(MessageBundle.getMessage("angal.lab.selectapatient"));
-			if(pat != null){
-				for (Patient elem : pat) {
-					if (lab.getPatient() != null && !insert) {
-						if (elem.getCode() == lab.getPatient().getCode()) {
-							patSelected=elem;
-						}
-					}
-					patientComboBox.addItem(elem);
-				}
-			}
-			if (patSelected!=null)
-				patientComboBox.setSelectedItem(patSelected);
-			
-			patientComboBox.addActionListener(new ActionListener() {
-				public void actionPerformed(ActionEvent arg0) {
-					if (patientComboBox.getSelectedIndex()>0) {
```

Comment:
removing this code means the imports for `ActionEvent` and `ActionListener` are no longer needed

Reply:
FWIW the two are called explicitely at:
![image](https://user-images.githubusercontent.com/2938553/104560555-547b5380-5646-11eb-8af7-f75e3e22bd2f.png)